cli/cli_app.go

```diff
@@ -274,16 +276,66 @@ func processHooks(hooks []config.Hook, terragruntOptions *options.TerragruntOpti
 	for _, curHook := range hooks {
 		allPreviousErrors := append(previousExecError, errorsOccurred...)
 		if shouldRunHook(curHook, terragruntOptions, allPreviousErrors...) {
+			var oldEnvMap map[string]string
+
 			terragruntOptions.Logger.Printf("Executing hook: %s", curHook.Name)
-			actionToExecute := curHook.Execute[0]
-			actionParams := curHook.Execute[1:]
-			possibleError := shell.RunShellCommand(terragruntOptions, actionToExecute, actionParams...)
+			if curHook.LoadEnvVars != nil {
+				terragruntOptions.Logger.Printf("Loading environment for hook: %s", curHook.Name)
+
+				actionToExecute := curHook.LoadEnvVars.Execute[0]
+				actionParams := curHook.LoadEnvVars.Execute[1:]
+				out, err := shell.RunShellCommandWithOutput(terragruntOptions, "", actionToExecute, actionParams...)
+				if err != nil {
+					terragruntOptions.Logger.Printf("Error running hook %s while loading environment variables with message: %s", curHook.Name, err.Error())
+					errorsOccurred = append(errorsOccurred, err)
+					continue
+				}
+				// parse out from script to map
+				m := make(map[string]string)
+				scanner := bufio.NewScanner(strings.NewReader(out.Stdout))
+				for scanner.Scan() {
+					txt := scanner.Text()
+					idx := strings.Index(txt, "=")
+					if idx > 0 {
+						key := strings.TrimSpace(txt[0:idx])
+						value := txt[idx+1 : len(txt)]
+						m[key] = value
+					} else {
+						terragruntOptions.Logger.Printf("Invalid key,value pair [%s]", txt)
+					}
+				}
+				targetEnvMap := terragruntOptions.Env
+				if curHook.LoadEnvVars.Transient {
+					oldEnvMap = terragruntOptions.Env
+					targetEnvMap := make(map[string]string)
+					for key, value := range terragruntOptions.Env {
+						targetEnvMap[key] = value
+					}
+					terragruntOptions.Env = targetEnvMap
+				}
+				if curHook.LoadEnvVars.Overwrite {
```

Comment:
Should this be an `else if`?

Reply:
No according to the matrix I added to my initial description of the PR, you can combine the flags `transient` and `overwrite`.